scripts/import_opentextbooks.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python
+import json
+import requests
+import time
+from typing import Any
+from openlibrary.core.imports import Batch
+from scripts.solr_builder.solr_builder.fn_to_cli import FnToCLI
+from openlibrary.config import load_config
+
+FEED_URL = 'https://open.umn.edu/opentextbooks/textbooks.json?'
+
+
+def get_feed():
+    """Fetches and returns the feed."""
+    r = requests.get(FEED_URL)
+    return r.json()
+
+
+def map_data(data) -> dict[str, Any]:
+    import_record = {}
+
+    if data.get("title"):
+        import_record["title"] = data["title"]
+
+    if data.get('ISBN10'):
+        import_record['isbn_10'] = data['ISBN10']
+
+    if data.get('ISBN13'):
+        import_record['isbn_13'] = data['ISBN13']
+
+    if data.get('language'):
+        import_record['languages'] = [data['language']]
+
+    if data.get('description'):
+        import_record['description'] = data['description']
+
+    if data.get('subjects'):
+        subjects = [
+            subject["name"] for subject in data['subjects'] if subject.get("name")
+        ]
+        if subjects:
+            import_record['subjects'] = subjects
+
+    if data.get('publishers') and data['publishers'][0]["name"]:
+        import_record['publishers'] = [data['publishers'][0]["name"]]
+
+    if data.get("copyright_year"):
+        import_record['publish_date'] = str(data["copyright_year"])
+
+    if data.get('id'):
+        import_record["identifiers"] = {'open_textbook_library': str(data['id'])}
+        import_record["source_records"] = ['open_textbook_library:%s' % data['id']]
```

Comment:
Move this to the top, and don't wrap in an `if`. We definitely want this to error if `id` is missing!